pkg/util/errors/errors.go

```diff
@@ -29,10 +29,16 @@ type Aggregate interface {
 // is itself an implementation of the error interface.  If the slice is empty,
 // this returns nil.
 func NewAggregate(errlist []error) Aggregate {
-	if len(errlist) == 0 {
+	errs := []error{}
+	for _, err := range errlist {
+		if err != nil {
```

Comment:
To be honest, as a user of the utility I would expect nil errors to be ignored, otherwise the result of an aggregation (eg. "nil nil nil") will be considered an error which is probably not what I expect.